GUI/src/cz/fidentis/gui/comparison_batch/BatchComparisonResults.java

```diff
@@ -2053,42 +2062,42 @@ private void arbitraryRadioButtonActionPerformed(java.awt.event.ActionEvent evt)
 
     private void normalSpinnerXStateChanged(javax.swing.event.ChangeEvent evt) {//GEN-FIRST:event_normalSpinnerXStateChanged
         if (normalSpinnerX.isEnabled() && !valuesModified) {
-            getContext().getArbitraryPlanePos().x = (float) normalSpinnerX.getValue();
+            getContext().setArbitraryPlanePos(new Vector3f((float)normalSpinnerX.getValue(), (float) normalSpinnerY.getValue(), (float) normalSpinnerZ.getValue()));
             GUIController.getSelectedProjectTopComponent().getViewerPanel_Batch().setPlaneNormal(getContext().getArbitraryPlanePos(), true);
         }
     }//GEN-LAST:event_normalSpinnerXStateChanged
 
     private void normalSpinnerYStateChanged(javax.swing.event.ChangeEvent evt) {//GEN-FIRST:event_normalSpinnerYStateChanged
         if (normalSpinnerY.isEnabled() && !valuesModified) {
-            getContext().getArbitraryPlanePos().y = (float) normalSpinnerY.getValue();
+            getContext().setArbitraryPlanePos(new Vector3f((float)normalSpinnerX.getValue(), (float) normalSpinnerY.getValue(), (float) normalSpinnerZ.getValue()));
             GUIController.getSelectedProjectTopComponent().getViewerPanel_Batch().setPlaneNormal(getContext().getArbitraryPlanePos(), true);
         }
     }//GEN-LAST:event_normalSpinnerYStateChanged
 
     private void normalSpinnerZStateChanged(javax.swing.event.ChangeEvent evt) {//GEN-FIRST:event_normalSpinnerZStateChanged
         if (normalSpinnerZ.isEnabled() && !valuesModified) {
-            getContext().getArbitraryPlanePos().z = (float) normalSpinnerZ.getValue();
+            getContext().setArbitraryPlanePos(new Vector3f((float)normalSpinnerX.getValue(), (float) normalSpinnerY.getValue(), (float) normalSpinnerZ.getValue()));
```

Comment:
dali by sa tieto veci prepisat tak, aby sa vzdy pri zmene nevytvaral novy Vector3f? ten even nastane x krat za sekundu to zaseka program. Podobne aj v ostatnych